interview-questions.md

```diff
@@ -7,44 +7,93 @@ Then, research the question to expand on your answer. Even if you feel you have
+    @post = BlogPost.find(params[:id])
+  end
 
+def destroy
+    @post = BlogPost.find(params[:id])
+    if @post.destroy
+      redirect_to blog_posts_path
+    else
+      # ---8)we are deleting, destroying and removing data from our @posts. We are sayign that if our post can not be delted then we must redirect to another path to our @post.
+
+
+Update and create routes also need params.
 
 3. Name three rails generator commands. What is created by each?
 
   Your answer:
 
+rails db:create - Is when you are creating database in rails you first input this command in the terminal and it creates the database you will work on
+rails s - you input rails s in the terminal and  you are running your rails server allowing your capabaility to see your rails app go live  on locahhost3000.
+rails db:migrate - it runs the single migrations you have not run yet. It updates the data base.
+
+
   Researched answer:
 
+rails db:create - creates the database
+rails s - begins the rails server by running the rails app
+rails db:migrate - this command updates different versions of your database to accomplish your task at different times. It updates the schema file that holds our created data that we can update, create, or destory data in.
 
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller (create, read, update, delete, index, new) methods perform?
 
 action: "GET"    location: /students          
+The name of the controller would that would be called for this route is create. Once you you create a template request and you go to /students you can see what ever content you put in the view/template request.
 
-action: "POST"   location: /students       
-
-action: "GET"    location: /students/new
+action: "POST"   location: /students        index method
 
-action: "GET"    location: /students/2  
+action: "GET"    location: /students/new     create method of new primary keys
+  
+action: "GET"    location: /students/2        only show one primary ket of data ex: student id:2
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit     edit the student id :2
 
-action: "PATCH"  location: /students/2      
+action: "PATCH"  location: /students/2       modify student id:2 key
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2         remove student id:2 
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+
+As a developer I can create a rails app
```

Comment:
These are great developer stories!  The main difference between a developer story and a user story is the user story will focus on what the user experience will be while the developer stories will be driven by specific programming tasks.